app/Http/Controllers/ScheduleController.php

```diff
@@ -335,137 +246,25 @@ public static function createScheduleEntries()
-                                                null);                  // new comment
-
-                // proceed with deletion
-                $entry->delete();
-
-            } 
-            // some new job type added - change entry
-            else 
-            {       
-                $jobtype = Jobtype::firstOrCreate(array('jbtyp_title'=>Input::get("jbtyp_title" . $counterHelper)));
-                $entry->jbtyp_id = $jobtype->id;
-
-                // log revision
-                /*
-                ScheduleController::logRevision($entry->getSchedule,    // schedule object
-                                                $entry,                 // entry object
-                                                "Dienst aktualisiert",      // action description
-                                                $entry->getPerson,      // old value
-                                                $entry->getPerson);     // new value
-                */
-                // add to new collection
-                $newEntries->add(ScheduleController::updateScheduleEntry($entry, $jobtype->id, $counterHelper));
-            }
-
-            // move to next input
-            $counterHelper++;
-        }
-
-        // At this point we changed all existing entries - have any new ones been added?
-
-        if ($numberOfSubmittedEntries > $counterHelper - 1) {
-            
-            // create some new fields
-            for ($i= $counterHelper; $i <= $numberOfSubmittedEntries; $i++) 
-            {
-                // skip empty fields, create new fields only if input not empty
-                if (!empty(Input::get("jbtyp_title" . $i))) 
-                {
-                    $jobtype = Jobtype::firstOrCreate(array('jbtyp_title'=>Input::get("jbtyp_title" . $i)));
-
-                    $newEntry = new ScheduleEntry;
-                    $newEntry->jbtyp_id = $jobtype->id;
-                    $newEntry->schdl_id = $scheduleId;
-
-                    // log revision
-                    ScheduleController::logRevision($newEntry->getSchedule, // schedule object
-                                                    $newEntry,              // entry object
-                                                    "Dienst erstellt",      // action description
-                                                    null,                   // old value
-                                                    null,                   // new value
-                                                    null,                   // old comment
-                                                    null);                  // new comment                   
-
-                    // add to new collection
-                    $newEntries->add(ScheduleController::updateScheduleEntry($newEntry, $jobtype->id, $i));
-                }
-            }
-        }
-
-        return $newEntries;
+        self::createShifts($scheduleId, false);
```

Comment:
Actually it should not be a call to `ShiftController@create`, as these kinds of methods are reserved for request (see [the Laravel Documentation](https://laravel.com/docs/5.4/controllers#resource-controllers)).

But you are right, there should be a call to `ShiftController@createShifts` or something like this.